### PR TITLE
End token handling

### PR DESCRIPTION
**Pull request by JensOgorek**
Adds fast '\0' handling without allocating memory. 
This is for #3

### CODE REVIEW

File: diff_list.md [hunk 1]
```diff
@@ -5,3 +5,6 @@ Changes in this fork
 
 owl_example_parser.sh:
     - Add an example script to generate the parser from the grammar
+
+src/x-tokenize.h:
+    - Add fast '\0' check to the tokenizer without allocating memory (Zauberzeug issue #99)
```
Comment (falkoschindler) [inline diff_list.md:10]:
```suggestion
    - Add fast '\0' check to the tokenizer without allocating memory (https://github.com/zauberzeug/lizard/issues/99)
```

File: src/x-tokenize.h [hunk 2]
```diff
@@ -236,6 +236,11 @@ static bool OWL_DONT_INLINE
 owl_default_tokenizer_advance(struct owl_default_tokenizer *tokenizer,
  struct owl_token_run **previous_run)
 {
+    // Issue: Zauberzeug/lizard#99
```
Comment (falkoschindler) [inline src/x-tokenize.h:239]:
```suggestion
    // https://github.com/zauberzeug/lizard/issues/99
```

**Review by falkoschindler — CHANGES_REQUESTED** [review]
Let's use links to allow jumping to the issue more quickly.
By the way: On GitHub a number like `#99` will link to issue 99 _in the same repository_. So it's better to use full links when it comes to third-party repos.

File: diff_list.md (hunk 1 above)
Comment (JensOgorek) [reply diff_list.md]: updated

File: src/x-tokenize.h (hunk 2 above)
Comment (JensOgorek) [reply src/x-tokenize.h]: updated

**Review by falkoschindler — APPROVED** [review]

**Comment by JensOgorek** [comment]
> Somehow you deleted gen_1-parse.sh...

yet it still shows up, that makes no sense

**Comment by JensOgorek** [comment]
it was just the +x that was gone

**Review by falkoschindler — APPROVED** [review]

**Pull request merged**